cpp/foxglove-websocket/src/serialization.cpp

```diff
@@ -103,17 +103,63 @@ void to_json(nlohmann::json& j, const Service& service) {
     {"id", service.id},
     {"name", service.name},
     {"type", service.type},
-    {"requestSchema", service.requestSchema},
-    {"responseSchema", service.responseSchema},
   };
+
+  if (service.request.has_value()) {
+    j["request"] = service.request.value();
+  }
+  if (service.response.has_value()) {
+    j["response"] = service.response.value();
+  }
+  if (service.requestSchema.has_value()) {
+    j["requestSchema"] = service.requestSchema.value();
+  }
+  if (service.responseSchema.has_value()) {
+    j["responseSchema"] = service.responseSchema.value();
+  }
 }
 
 void from_json(const nlohmann::json& j, Service& p) {
   p.id = j["id"].get<ServiceId>();
   p.name = j["name"].get<std::string>();
   p.type = j["type"].get<std::string>();
-  p.requestSchema = j["requestSchema"].get<std::string>();
-  p.responseSchema = j["responseSchema"].get<std::string>();
+
+  {
+    const auto it = j.find("request");
+    p.request = it == j.end() ? std::optional<ServiceRequestDefinition>(std::nullopt)
+                              : std::make_optional(it->get<ServiceRequestDefinition>());
+  }
```

Comment:
nit: possible cleanup
```suggestion
  if (const auto it = j.find("request"); it != j.end()) {
    p.request = it->get<ServiceRequestDefinition>();
  }
```

Reply:
much better, thanks!